django-rgd-imagery/client/rgd_imagery_client/plugin.py

```diff
@@ -344,3 +344,69 @@ def create_processed_image(
 
         r.raise_for_status()
         return r.json()
+
+    def get_leaflet_tile_source(
+        self,
+        image_id: Union[str, int],
+        band: int = None,
+        palette: str = None,
+        vmin: Union[float, int] = None,
+        vmax: Union[float, int] = None,
+        nodata: Union[float, int] = None,
+        **kwargs,
+    ):
+        """Generate an ipyleaflet TileLayer for the given Image.
+
+        Parameters
+        ----------
+        image_id : Union[str, int]
+            The image ID to serve tiles from
+
+        **kwargs
+            All additional keyword arguments are passed to TileLayer
+
+        Return
+        ------
+        ipyleaflet.TileLayer
+
+        """
+        # Safely import ipyleaflet
+        try:
+            from ipyleaflet import TileLayer
+        except ImportError:
+            raise ImportError('Please install `ipyleaflet` and `jupyter`.')
```

Comment:
`ipyleaflet` isn't listed anywhere in `setup.py` as an requirement. It probably should be, if nothing else under an extra requirement.